### ISSUE TITLE
Guide to use uses_learning_phase and training parameter

### ISSUE THREAD

**Issue by rudaoshi** [fchollet/keras#6489]
I noticed that a training parameter had been added to the __call__ and call function of layers. 

However, I used to use uses_learning_phase to decide wether the model is in training or prediction. What's the reason of introducing the new training parameter ?
When I develop new layer, which one should be used?

**Comment by kgrm** [comment]
The parameter is used for layers that have different behaviour in training and testing/evaluation/use phases, such as `Dropout` and `BatchNormalization`. See their code for reference.

**Comment by rudaoshi** [comment]
But since I can know that by uses_learning_phase, why add another parameter?

**Comment by Darthholi** [comment]
Also not clear to me

**Comment by chrishkchris** [comment]
not clear how to use
